client/jetpack-app/plans/main.tsx

```diff
@@ -118,6 +118,7 @@ const JetpackAppPlans: React.FC< JetpackAppPlansProps > = ( { paidDomainName, or
 						hidePlanTypeSelector
 						hidePlansFeatureComparison
 						setSiteUrlAsFreeDomainSuggestion={ setSiteUrlAsFreeDomainSuggestion }
+						flowName="onboarding-jetpack-app"
```

Comment:
Since technically there is no flow involved here, what you can do is use the intent property above instead.

Just feed the intent into the useModalResolutionCallback hook and check that, also make sure to write a unit test below.

https://github.com/Automattic/wp-calypso/blob/ad35534162a9a39f823b260d4437550bee614ce6/client/my-sites/plans-features-main/components/plan-upsell-modal/test/plan-upsell-modal.test.tsx

Reply:
Thanks for the suggestion! 🙇 Yes, I agree that this is not technically a flow. I also added 2 tests to verify two main cases: a paid domain should trigger this dialog while a free one should not. 

I see that there's CI failure but it doesn't look to be related to the changes I made so I'm giving this PR back to review.
<img width="676" alt="image" src="https://github.com/Automattic/wp-calypso/assets/4062343/a2130629-a3e2-4867-bc83-2cab807654cc">